src/parser/jobs/pld/modules/Requiescat.tsx

```diff
@@ -23,223 +19,109 @@ const SEVERITIES = {
+		casts: 0,
+		buffs: 0,
 	}
 
-	protected override init() {
-		this.addEventHook('cast', {by: 'player'}, this.onCast)
-		this.addEventHook(
-			'applybuff',
-			{by: 'player', abilityId: STATUSES.REQUIESCAT.id},
-			this.onApplyRequiescat,
-		)
-		this.addEventHook(
-			'removebuff',
-			{by: 'player', abilityId: STATUSES.REQUIESCAT.id},
-			this.onRemoveRequiescat,
-		)
-		this.addEventHook('complete', this.onComplete)
-	}
-
-	private onCast(event: CastEvent) {
-		const actionId = event.ability.guid
-
-		if (actionId === ACTIONS.ATTACK.id) {
-			return
-		}
-
-		if (actionId === ACTIONS.REQUIESCAT.id) {
-			// Add new cast to the list
-			const reqState = new RequiescatState(event.timestamp)
-			const reqEnd = event.timestamp + STATUSES.REQUIESCAT.duration
-
-			if (reqEnd >= this.parser.fight.end_time) {
-				// If the requiescat overshoots the end of the fight, we know ahead of time it'll be a rush
-				reqState.isRushing = true
-			} else {
-				// Otherwise, wait for the expected end time and check invuln status
-				this.addTimestampHook(reqEnd, ({timestamp}) => {
-					const epochTimestamp = this.parser.fflogsToEpoch(timestamp)
-					reqState.isRushing = false
-						|| this.invulnerability.isActive({timestamp: epochTimestamp, types: ['invulnerable']})
-						|| this.invulnerability.isActive({timestamp: epochTimestamp, types: ['untargetable']})
-				})
-			}
-
-			this.requiescats.push(reqState)
-		}
+	override initialise() {
+		super.initialise()
+
+		this.addEvaluator(new ExpectedActionsEvaluator({
+			expectedActions: [
+				{action: [this.data.actions.HOLY_SPIRIT, this.data.actions.HOLY_CIRCLE], expectedPerWindow: 3},
+				{action: this.data.actions.CONFITEOR, expectedPerWindow: 1},
+			],
+			suggestionIcon: this.data.actions.HOLY_SPIRIT.icon,
+			suggestionContent: <Trans id="pld.requiescat.suggestions.wrong-gcd.content">
+				GCDs used during <DataLink action="REQUIESCAT" /> should consist of 3-4 uses of <DataLink action="HOLY_SPIRIT" /> (or
+				multi-hit <DataLink action="HOLY_CIRCLE" />) and 1 use of <DataLink action="CONFITEOR" /> for optimal damage.
+			</Trans>,
+			suggestionWindowName: <DataLink action="REQUIESCAT" showIcon={false} />,
+			severityTiers: {
+				1: SEVERITY.MEDIUM,
+				2: SEVERITY.MAJOR,
+			},
+			adjustCount: this.adjustExpectedActionsCount.bind(this),
+		}))
 
-		const lastRequiescat = this.lastRequiescat
+		this.addEventHook({type: 'action', source: this.parser.actor.id, action: this.data.actions.REQUIESCAT.id}, () => this.requiescatUsages.casts++)
+		this.addEventHook({type: 'statusApply', target: this.parser.actor.id, status: this.data.statuses.REQUIESCAT.id}, () => this.requiescatUsages.buffs++)
 
-		// If we're still in the considered window, log our actions to it
-		if (lastRequiescat != null && lastRequiescat.end == null) {
-			lastRequiescat.rotation.push(event)
-		}
+		this.addEvaluator(new RequiescatUsageEvaluator(this.requiescatUsages, this.data.actions.REQUIESCAT.icon))
```

Comment:
mutable cross-object references aaaaaaaaaaaaaaaa
thanks i hate it

not sure i know a better way without thinking about it more tho